### PR TITLE
1886 importer unit for fraction not recognized

### LINKED ISSUE #1886: Data importer: Unit for "Fraction" is not recognized
The default unit for the dimension "Fraction" is an empty string, which is not recognized by the importer.

image
[core-importer-fraction-unit.zip](https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/files/10579777/core-importer-fraction-unit.zip)

### PR DESCRIPTION
**Pull request by georgeDaskalakis**
closes #1886

### CODE REVIEW

File: src/OSPSuite.Infrastructure.Import/Core/DataSource.cs [hunk 1]
```diff
@@ -263,16 +263,10 @@ private ParseErrors validateUnitsSupportedAndSameDimension(Cache<string, ColumnI
                   //if unit comes from a column
                   if (column.Key.Column.Dimension == null)
                   {
-                     var firstNonEmptyUnit = column.Value.FirstOrDefault(x => !string.IsNullOrEmpty(x.Unit));
+                     var firstValueWithNonEmptyUnit = column.Value.FirstOrDefault(x => !string.IsNullOrEmpty(x.Unit));
+                     var firstNonEmptyUnit = firstValueWithNonEmptyUnit == null ? "" : firstValueWithNonEmptyUnit.Unit;
```
Comment (msevestre) [inline src/OSPSuite.Infrastructure.Import/Core/DataSource.cs:267]: FYI you can write it as 

`var firstNonEmptyUnit = firstValueWithNonEmptyUnit?.Unit??"";`

File: tests/OSPSuite.Infrastructure.Tests/Import/DataSourceSpecs.cs [hunk 2]
```diff
@@ -164,6 +164,11 @@ protected override void Context()
          sut = new DataSource(_fakedImporter);
          sut.DataSets.Add("sheet1", _fakeDataSet);
       }
+
+      protected string getFirstMeasurementUnit()
+      {
+         return sut.DataSets.KeyValues.First().Value.Data.First().Data.ToList()[1].Value[1].Unit;
```
Comment (msevestre) [inline tests/OSPSuite.Infrastructure.Tests/Import/DataSourceSpecs.cs:170]: This structure lol. Value.Data.Data.Value...

**Review by msevestre — APPROVED** [review]

File: src/OSPSuite.Infrastructure.Import/Core/DataSource.cs (hunk 1 above)
Comment (georgeDaskalakis) [reply src/OSPSuite.Infrastructure.Import/Core/DataSource.cs:267]: I like the suggestion. Implemented.

File: tests/OSPSuite.Infrastructure.Tests/Import/DataSourceSpecs.cs (hunk 2 above)
Comment (georgeDaskalakis) [reply tests/OSPSuite.Infrastructure.Tests/Import/DataSourceSpecs.cs:170]: @msevestre Yes I know - and totally agree. The only reason I would leave this here is that it is only the test and in the actual code it is never called like taht - the unit is deduced further down in the processesing.

**Pull request merged**